### PR TITLE
netdev/lora: fix size of NETOPT_RX_SYMBOL_TIMEOUT

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
The definition of NETOPT_RX_SYMBOL_TIMEOUT [states](https://doc.riot-os.org/group__net__netopt.html#gga19e30424c1ab107c9c84dc0cb29d9906aadd8dd23500923fd80a3f5e4c452fdfa) that the size should be `uint8_t` instead of `uint16_t`.

The documentation makes sense IMO:
- The rx symbol timeout register is 1 byte in SX127x
- Larger values are not suitable for common applications such as LoRaWAN.

It's guaranteed that GNRC LoRaWAN won't pick a value greater than UINT8_MAX. Although it's defined as `uint16_t` in semtech loramac, in practice this value won't be bigger than UINT8_MAX neither because it depends on the configuration values. A proof of that is the fact sx127x runs with semtech-loramac.
I just added an assertion to the pkg just in the rare case this changes in future semtech-loramac implementations.

### Testing procedure
GNRC LoRaWAN and semtech-loramac should compile and work as always.

### Issues/PRs references
Detected during #16579

### CODE REVIEW

File: drivers/sx127x/sx127x_netdev.c [hunk 1]
```diff
@@ -444,8 +444,8 @@ static int _set(netdev_t *netdev, netopt_t opt, const void *val, size_t len)
         return sizeof(netopt_enable_t);
 
     case NETOPT_RX_SYMBOL_TIMEOUT:
-        assert(len <= sizeof(uint16_t));
-        sx127x_set_symbol_timeout(dev, *((const uint16_t *)val));
+        assert(len <= sizeof(uint8_t));
```
Comment (fjmolinas) [inline drivers/sx127x/sx127x_netdev.c:447]: Huh? `assert(len <= 1)? symbol timeout is max 1 symbol?

**Comment by fjmolinas** [comment]
> neither because it depends on the configuration values. 

Can you explain more?

File: drivers/sx127x/sx127x_netdev.c (hunk 1 above)
Comment (leandrolanzieri) [reply drivers/sx127x/sx127x_netdev.c:447]: I think this is just making sure that the value is `uint8_t`, as it is checking `len`, not `val`.

Comment (jia200x) [reply drivers/sx127x/sx127x_netdev.c:447]: > I think this is just making sure that the value is uint8_t, as it is checking len, not val.

Yes. This just checks that the data structure that holds `val` is not greater than 1 byte (uint8t_t)

**Comment by jia200x** [comment]
> Can you explain more?

The number of symbols used for RX Timeout is given by the following function in Semtech LoRaMAC (windowTimeout):
```c
void RegionCommonComputeRxWindowParameters( double tSymbol, uint8_t minRxSymbols, uint32_t rxError, uint32_t wakeUpTime, uint32_t* windowTimeout, int32_t* windowOffset )
{
    *windowTimeout = MAX( ( uint32_t )ceil( ( ( 2 * minRxSymbols - 8 ) * tSymbol + 2 * rxError ) / tSymbol ), minRxSymbols ); // Computed number of symbols
    *windowOffset = ( int32_t )ceil( ( 4.0 * tSymbol ) - ( ( *windowTimeout * tSymbol ) / 2.0 ) - wakeUpTime );
}
```

`minRxSymbols` (mapped to `CONFIG_LORAMAC_DEFAULT_MIN_RX_SYMBOLS` and `rxError` (mapped to `CONFIG_LORAMAC_DEFAULT_SYSTEM_MAX_RX_ERROR`) should be configured in a way that the windowTimeout doesn't exceed 255. Otherwise sx127x radios won't work and even if the NETOPT returns an error, the stack has no way to recover.

The default values (12 for minRxSumbol and 50 for rxError) ensure this is true.

For GNRC LoRaWAN we are only using one value (`GNRC_LORAWAN_MIN_SYMBOLS_TIMEOUT`) . This value is lmited by Kconfig to less than 255.

BTW, maybe I can remove this config from GNRC LoRaWAN since it's already included in the generic LoRaMAC configurations.

**Comment by jia200x** [comment]
technically this is a bug. We should IMO try to backport it

**Comment by fjmolinas** [comment]
@jia200x I don't have access to a gateway, can you post debug output? I can ACK afterwards.

**Comment by fjmolinas** [comment]
I verified that I can still communicate among radios.

**Comment by jia200x** [comment]
here's the test output with `semtech-loramac`:

```
2021-07-06 11:52:03,582 # loramac join otaa
2021-07-06 11:52:11,949 # Join procedure succeeded!
loramac tx hola
2021-07-06 11:55:52,580 # loramac tx hola
2021-07-06 11:55:56,089 # Received ACK from network
2021-07-06 11:55:56,091 # Message sent with success
> loramac tx test
2021-07-06 11:56:12,476 # loramac tx test
2021-07-06 11:56:16,148 # Data received: RIOT, port: 2
2021-07-06 11:56:16,150 # Received ACK from network
2021-07-06 11:56:16,155 # Message sent with success
> reboot
2021-07-06 11:56:27,659 # reboot
2021-07-06 11:56:27,731 # main(): This is RIOT! (Version: 2021.07-devel-448-g5ef4c-pr/lora/fix_rx_timeout_size)
2021-07-06 11:56:27,733 # All up, running the shell now
> 
```

**Review by fjmolinas — APPROVED** [review]
ACK!

**Comment by jia200x** [comment]
my bad... I just had a corrupted brain cache and discovered 2 things:

1. The SX127x DOES accept an MSB value for RX symbol timeout.
2. This PR was also wrong, because it's now setting the symbol timeout in the LSB.

The consequences of 2 are not enough to make the LoRaWAN Release tests fail. Therefore we should try to add regression tests there.

I propose to simply revert this PR and just fix the documentation. We should of course backport that. I will take care of that.

**Pull request merged**